### PR TITLE
CAV-189 - cip phone number verification metrics

### PR DESCRIPTION
**Pull request by mar36tin**

### CODE REVIEW

File: app/uk/gov/hmrc/cipphonenumbervalidation/metrics/MetricsService.scala [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import com.codahale.metrics.Counter
+import com.google.inject.Inject
+import com.kenshoo.play.metrics.Metrics
+
+import javax.inject.Singleton
+
+@Singleton
+class MetricsService @Inject() (metrics: Metrics){
+
+
```
Comment (johno1985) [inline app/uk/gov/hmrc/cipphonenumbervalidation/metrics/MetricsService.scala:28]: a bit pedantic I know...

File: test/uk/gov/hmrc/cipphonenumbervalidation/service/ValidateServiceSpec.scala [hunk 2]
```diff
@@ -84,6 +86,7 @@ class ValidateServiceSpec extends AnyWordSpec
   }
 
   trait SetUp {
-    val validateService = new ValidateService(PhoneNumberUtil.getInstance())
+    lazy val mockMetricsServer = mock[MetricsService]
+    val validateService = new ValidateService(PhoneNumberUtil.getInstance(), mockMetricsServer)
```
Comment (johno1985) [inline test/uk/gov/hmrc/cipphonenumbervalidation/service/ValidateServiceSpec.scala:90]: should we add some verifications to the unit tests:

`mockMetricsServer.recordMetric("metricName") was called`

Comment (brdgeth) [reply test/uk/gov/hmrc/cipphonenumbervalidation/service/ValidateServiceSpec.scala:90]: i agree, I think we said we should do this when we were refining it as well.

Comment (mar36tin) [reply test/uk/gov/hmrc/cipphonenumbervalidation/service/ValidateServiceSpec.scala:90]: good idea guys yes

**Review by johno1985 — APPROVED** [review]

**Pull request merged**